chains/manager.go

```diff
@@ -622,26 +656,69 @@ func (m *manager) createSnowmanChain(
 	// VM uses this channel to notify engine that a block is ready to be made
 	msgChan := make(chan common.Message, defaultChannelSize)
 
-	// Initialize the VM
-	chainConfig := m.getChainConfig(ctx.ChainID)
-	if err := vm.Initialize(ctx, vmDBManager, genesisData, chainConfig.Upgrade, chainConfig.Config, msgChan, fxs); err != nil {
-		return nil, err
-	}
-
 	// Passes messages from the consensus engine to the network
 	sender := sender.Sender{}
-	err = sender.Initialize(
+	if err := sender.Initialize(
 		ctx,
+		m.MsgCreator,
 		m.Net,
 		m.ManagerConfig.Router,
 		m.TimeoutManager,
 		consensusParams.Namespace,
 		consensusParams.Metrics,
-	)
-	if err != nil {
+		m.AppGossipValidatorSize,
+		m.AppGossipNonValidatorSize,
+		m.GossipAcceptedFrontierSize,
+	); err != nil {
 		return nil, fmt.Errorf("couldn't initialize sender: %w", err)
 	}
 
+	// first vm to be init is P-Chain once, which provides validator interface to all ProposerVMs
```

Comment:
Check how we can use our static validator list for the validator interface for the proposer VMs.